src/core/meshcop/commissioner.cpp

```diff
@@ -393,16 +393,21 @@ void Commissioner::UpdateJoinerExpirationTimer(void)
     // Check if timer should be set for next Joiner.
     for (size_t i = 0; i < OT_ARRAY_LENGTH(mJoiners); i++)
     {
+        int32_t diff;
+
+        if (!mJoiners[i].mValid)
         {
-            if (!mJoiners[i].mValid)
-            {
-                continue;
-            }
+            continue;
+        }
 
-            if (mJoiners[i].mExpirationTime - now < nextTimeout)
-            {
-                nextTimeout = mJoiners[i].mExpirationTime - now;
-            }
+        diff = TimerMilli::Diff(now, mJoiners[i].mExpirationTime);
+        if (diff <= 0)
```

Comment:
not sure but is there any chance for some Joiner who is expired but not removed when entering this function?

Reply:
Yes, as you see this function re-get the now value. However, I admit, maybe could enhance the implementation in separate PR to merge `HandleJoinerExpirationTimer` and `UpdateJoinerExpirationTimer`.